File: shared/src/main/java/ru/privatenull/pnauth/command/AuthPlatformBridge.java
package ru.privatenull.pnauth.command;

import java.util.UUID;

public interface AuthPlatformBridge {
    AuthPlatformBridge NONE = new AuthPlatformBridge() {
        @Override
        public void authenticated(UUID uniqueId) {
        }

        @Override
        public void loggedOut(UUID uniqueId) {
        }

        @Override
        public void accountDeleted(UUID uniqueId) {
        }
    };

    void authenticated(UUID uniqueId);

    void loggedOut(UUID uniqueId);

    void accountDeleted(UUID uniqueId);

    default void authenticated(String username) {
    }

    default void accountDeleted(String username) {
    }

    /** Sends a pre-rendered announcement to every player connected to the proxy. */
    default void broadcast(String message) {
    }

    default void apply(CommandEffect effect, UUID uniqueId) {
        switch (effect) {
            case AUTHENTICATED -> authenticated(uniqueId);
            case LOGGED_OUT -> loggedOut(uniqueId);
            case ACCOUNT_DELETED -> accountDeleted(uniqueId);
            case NONE -> {
            }
        }
    }
}

File: shared/src/main/kotlin/ru/privatenull/pnauth/command/AuthPlatformBridge.kt
package ru.privatenull.pnauth.command

import java.util.UUID

interface AuthPlatformBridge {
    fun authenticated(uniqueId: UUID)

    fun loggedOut(uniqueId: UUID)

    fun accountDeleted(uniqueId: UUID)

    fun authenticated(username: String) {}

    fun accountDeleted(username: String) {}

    /** Sends a pre-rendered announcement to every player connected to the proxy. */
    fun broadcast(message: String) {}

    fun apply(effect: CommandEffect, uniqueId: UUID) {
        when (effect) {
            CommandEffect.AUTHENTICATED -> authenticated(uniqueId)
            CommandEffect.LOGGED_OUT -> loggedOut(uniqueId)
            CommandEffect.ACCOUNT_DELETED -> accountDeleted(uniqueId)
            CommandEffect.NONE -> {}
        }
    }

    companion object {
        @JvmField
        val NONE: AuthPlatformBridge = object : AuthPlatformBridge {
            override fun authenticated(uniqueId: UUID) {}
            override fun loggedOut(uniqueId: UUID) {}
            override fun accountDeleted(uniqueId: UUID) {}
        }
    }
}
